docs/docs/developers/development/dialogs/index.md

```diff
@@ -64,4 +70,105 @@ It will result with following in UI dialog:
 
 ![Dialog example tab2](dialog-example-tab2.png)
 
-Explore available fields and options.
\ No newline at end of file
+## Data structure
+All fields within the dialog have the `name` property, which defines the property’s name, under which values are saved in JCR. What is significant, the data is flattened, which means it does not matter if a particular field is in a nested structure (like tabs) or not. Properties are always saved under the same node.
```

Comment:
- mention that name can be relative path like 'subnode/property-name'
- mention about multifield - in this case values are saved in structure